device/thunder_ripple_sdk/src/client/thunder_async_client.rs

```diff
@@ -367,13 +367,12 @@ impl ThunderAsyncClient {
                                 let response = ThunderAsyncResponse::new_error(request.id,e.clone());
                                 match e {
                                     RippleError::ServiceNotReady => {
-                                        info!("prepare request failed for request {:?}", request);
+                                        info!("Thunder Service not ready, request added to pending list{:?}", request);
```

Comment:
Where is it being added? Or does it remain in the pending list upon error?

Reply:
I believe there is still a bug in this area. The second request is not being added to the pending list.